### PR TITLE
More robust mechanism for importing user code

### PR DESCRIPTION
**Pull request by alexsparrow**

### CODE REVIEW

File: src/quartic/pipeline/runner/cli.py [hunk 1]
```diff
@@ -32,15 +32,15 @@ def parse_args(argv):
 def run_user_code(f, exception_file):
     try:
         return f()
-    except ModuleNotFoundError as e:
-        raise ModuleNotFoundException(e.name)
+    except ModuleNotFoundException as e:
+        raise ModuleNotFoundException(e.module)
```
Comment (choliver) [inline src/quartic/pipeline/runner/cli.py]: Is this weird exception conversion wrong?

File: src/quartic/common/utils.py [hunk 2]
```diff
@@ -1,48 +1,42 @@
...
-        else:
-            continue
-    return files
+def load_package(package_dir):
+    # First load the root package
+    package_name = os.path.basename(package_dir)
+    init_py = os.path.join(package_dir, "__init__.py")
+    if not os.path.exists(init_py):
+        raise QuarticException("Package {} is missing __init__.py".format(package_dir))
+    mspec = importlib.util.spec_from_file_location(os.path.dirname(package_dir), init_py)
```
Comment (choliver) [inline src/quartic/common/utils.py:20]: Pull duplicate code into helper function?

File: src/quartic/common/utils.py [hunk 3]
```diff
@@ -1,48 +1,42 @@
...
-            files += find_python_files(a)
-        if a.endswith(".py"):
-            files.append(a)
-    if files:
-        return files
-    else:
-        import sys
-        print("No files found for current config. Bailing.") #TODO - log this scenario
-        sys.exit(1)
+    # This is a bit of a hack to make sure the submodules can find this package
```
Comment (choliver) [inline src/quartic/common/utils.py]: Probably add an illustrative code example, plus any other test cases for scenarios you're worried about.

File: src/quartic/pipeline/runner/cli.py (hunk 1 above)
Comment (alexsparrow) [reply src/quartic/pipeline/runner/cli.py]: I've reverted to what it was before. If we see again let's fix.

**Review by choliver — APPROVED** [review]

**Pull request merged**